src/main/scala/com/mwz/sonar/scala/scapegoat/ScapegoatReportParser.scala

```diff
@@ -32,9 +32,21 @@ trait ScapegoatReportParserAPI {
 final class ScapegoatReportParser extends ScapegoatReportParserAPI {
   private[this] val AllDotsButLastRegex = raw"\.(?=.*\.)".r
 
-  /** Replaces all dots (.) except the last one in a scapegoat path with slashes (/) */
+  /**
+   * Replaces all dots '.' except the last one in a scapegoat path with slashes '/'
+   * while keeping valid directories which contain '.' in their name.
+   */
   private[scapegoat] def replaceAllDotsButLastWithSlashes(path: String): String =
-    AllDotsButLastRegex.replaceAllIn(target = path, replacement = "/")
+    if (path.startsWith(".")) {
```

Comment:
I can't confirm it, neither test it _(cause I'm on my cell phone right now)_.  
But, I think this isn't working the way you think.
For example, I think this file will make it fail: `target.scala.12..file.Scala`

Reply:
I'll add some tests with extra dots in the middle of the path, but I think it's rather uncommon to have Scala source files hidden, i.e. starting with a `.`, so I think we can safely ignore this case.

Reply:
Hidden files probably don't, but what about paths like `scala-2.12` ?

Reply:
yeah, so your example would work since the implementation checks first if `scala-2` exists in the file system and if it doesn't instead of resolving `12` as a subdirectory it concatenates it with a `.` and the previous path before checking again for existence in the file system - I'll add a unit test to cover that.